man/man8/zfs.8

```diff
@@ -3928,27 +3940,45 @@ descendent file systems.
 .It Xo
 .Nm
 .Cm diff
-.Op Fl FHt
-.Ar snapshot Ar snapshot Ns | Ns Ar filesystem
+.Op Fl FHta
+.Ar snapshot Ns | Ns Ar bookmark
+.Op Ar snapshot Ns | Ns Ar filesystem
 .Xc
-Display the difference between a snapshot of a given filesystem and another
-snapshot of that filesystem from a later time or the current contents of the
-filesystem.
-The first column is a character indicating the type of change, the other columns
-indicate pathname, new pathname
+Display the differences in a filesystem between two points in time.
+The earlier point in time is specified as the first argument, which can be a
+snapshot or a bookmark.
+The later point in time is specified by the second argument, which can be a
+snapshot, a filesystem, or absent.
+If the second argument is absent, it is assumed to be the current state of
+the first argument's filesystem.
+.sp
```

Comment:
You're going to want to replace `.sp` with `.Pp` for paragraph breaks everywhere.